toolkit/tools/imagegen/configuration/parse_partition_test.go

```diff
@@ -0,0 +1,126 @@
+// Copyright Microsoft Corporation.
+// Licensed under the MIT License.
+
+package configuration
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	InvalidOnDiskPartitionCommand = "part / --fstype=ext4 --size=800 --ondisk="
+	InvalidFstypePartitionCommand = "part / --fstype=     --size=800 --ondisk=/dev/sda"
+	InvalidSizePartitionCommand   = "part / --fstype=ext4 --size=abcd --ondisk=/dev/sda"
+
+	ValidPartitionCommand1 = "part / --fstype=ext4 --size=800 --ondisk=/dev/sda"
+	ValidPartitionCommand2 = "part biosboot --fstype=biosboot --size=8 --ondisk=/dev/sda"
+
+	validTestParserDisk_OnePartition = Disk{
+		PartitionTableType: "gpt",
+		TargetDisk: TargetDisk{
+			Type:  "path",
+			Value: "/dev/sda",
+		},
+		Partitions: []Partition{
+			{
+				ID:     "rootfs",
+				Start:  1,
+				End:    801,
+				FsType: "ext4",
+			},
+		},
+	}
+
+	validTestParserDisk_TwoPartitions = Disk{
+		PartitionTableType: "gpt",
+		TargetDisk: TargetDisk{
+			Type:  "path",
+			Value: "/dev/sda",
+		},
+		Partitions: []Partition{
+			{
+				ID: "boot",
+				Flags: []PartitionFlag{
+					"bios_grub",
+				},
+				Start:  1,
+				End:    9,
+				FsType: "fat32",
+			},
+			{
+				ID:     "rootfs",
+				Start:  9,
+				End:    809,
+				FsType: "ext4",
+			},
+		},
+	}
+
+	validTestParserPartitionSettings = []PartitionSetting{
+		{
+			ID:              "boot",
+			MountPoint:      "",
+			MountIdentifier: GetDefaultMountIdentifier(),
+		},
+		{
+			ID:              "rootfs",
+			MountPoint:      "/",
+			MountIdentifier: GetDefaultMountIdentifier(),
+		},
+	}
+)
+
+func TestShouldFailParsingInvalidOnDiskPartitionCommand(t *testing.T) {
+	initializePrerequisitesForParser()
+
+	err := parsePartitionFlags(InvalidOnDiskPartitionCommand)
+	assert.Error(t, err)
+	assert.Equal(t, "--ondisk/--ondrive must not be empty", err.Error())
```

Comment:
Couldn't we define these error messages as `const` in their package and reuse those `consts` here?